### PR TITLE
Added funcs.go to templateutil to collect useful functions for Go templates

### PR DESCRIPTION
**Pull request by viktorbenei**
Commonly used functions for templates; functions that you can use as `template.FuncMap` functions.

Copied from https://github.com/bitrise-io/gotgen/blob/125332508853e1117c5e8dae9096fa69f0801e7d/cmd/generate.go#L106

I think it'd make sense to collect these functions here, in `go-utils`, so that these can then be used in projects which use `go-utils/templateutil`.

### CODE REVIEW

**Comment by viktorbenei** [comment]
Strange.. If I run `errcheck "-asserts=true" "-blank=true" "-verbose" "github.com/bitrise-io/go-utils/templateutil"` in any way I can't reproduce this issue..

**Comment by BirmacherAkos** [comment]
If you remove `the github.com/pkg/errors` package from your local computer it will fail for you too.
I've added it to the dep list, it should be fixed now.

**Comment by viktorbenei** [comment]
Wow that's a fantastic catch, thanks @BirmacherAkos !

File: templateutil/funcs_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+	"github.com/bitrise-io/go-utils/envutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetenv(t *testing.T) {
+	testEnvKey := "KEY_TestGetenv"
+
+	// no env set yet, empty value should be returned
+	require.Equal(t, "", Getenv(testEnvKey))
```
Comment (godrei) [inline templateutil/funcs_test.go]: pls do not use any testing framework for new tests, we want to deintroduce them.

File: bitrise.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
-format_version: 5
+format_version: "5"
 default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
-
-
+project_type: ""
 workflows:
   test:
     steps:
```
Comment (BirmacherAkos) [inline bitrise.yml]: Could you please update the go-tests for the newer solution
Example:
https://github.com/bitrise-io/go-xcode/blob/master/bitrise.yml#L17-L20

**Comment by viktorbenei** [comment]
To answer the question 

> other than that, why do we duplicate these functions across envutil and templateutil packages, related pr: #85?

I didn't want to base this PR on the other one. But I agree that once that's merged it'd make sense to use those functions.

File: bitrise.yml (hunk 2 above)
Comment (viktorbenei) [reply bitrise.yml]: I checked it but the same steps are used here too, the only difference is that in `go-list` a few `exclude`s are defined here.

**Comment by viktorbenei** [comment]
@BirmacherAkos updated + commented ( https://github.com/bitrise-io/go-utils/pull/88#discussion_r275404049 )

**Comment by viktorbenei** [comment]
I'll close this PR as in its current form I agree it doesn't really add anything new/useful. I might have some more time in the future to add a few more functions and some template function related utility methods from `gotgetn` e.g. the arithmetic functions https://github.com/bitrise-io/gotgen/blob/e4b5c30d4fe7f6f04c98c21d3d0aa71540ea0962/cmd/generate.go#L138 as well as the "Template functions constructor" func https://github.com/bitrise-io/gotgen/blob/e4b5c30d4fe7f6f04c98c21d3d0aa71540ea0962/cmd/generate.go#L105

File: bitrise.yml (hunk 2 above)
Comment (BirmacherAkos) [reply bitrise.yml]: Yeah, you are right. The exclude list misled me for some reason

**Pull request closed without merging**